syntax = "proto3";

package contracts.services.gateway.operations;

import "contracts/services/gateway/operations/rpc_get_operation.proto";
import "contracts/services/gateway/operations/rpc_get_operations.proto";
import "contracts/services/gateway/operations/rpc_get_operation_receipt.proto";
import "contracts/services/gateway/operations/rpc_get_operations_summary.proto";
import "contracts/services/gateway/operations/rpc_make_fee_operation.proto";
import "contracts/services/gateway/operations/rpc_make_top_up_operation.proto";
import "contracts/services/gateway/operations/rpc_make_cashback_operation.proto";
import "contracts/services/gateway/operations/rpc_make_purchase_operation.proto";
import "contracts/services/gateway/operations/rpc_make_transfer_operation.proto";
import "contracts/services/gateway/operations/rpc_make_bill_payment_operation.proto";
import "contracts/services/gateway/operations/rpc_make_cash_withdrawal_operation.proto";

service OperationsGatewayService {
  rpc GetOperation (GetOperationRequest) returns (GetOperationResponse);
  rpc GetOperations (GetOperationsRequest) returns (GetOperationsResponse);
  rpc GetOperationReceipt (GetOperationReceiptRequest) returns (GetOperationReceiptResponse);
  rpc GetOperationsSummary (GetOperationsSummaryRequest) returns (GetOperationsSummaryResponse);
  rpc MakeFeeOperation (MakeFeeOperationRequest) returns (MakeFeeOperationResponse);
  rpc MakeTopUpOperation (MakeTopUpOperationRequest) returns (MakeTopUpOperationResponse);
  rpc MakeCashbackOperation (MakeCashbackOperationRequest) returns (MakeCashbackOperationResponse);
  rpc MakePurchaseOperation (MakePurchaseOperationRequest) returns (MakePurchaseOperationResponse);
  rpc MakeTransferOperation (MakeTransferOperationRequest) returns (MakeTransferOperationResponse);
  rpc MakeBillPaymentOperation (MakeBillPaymentOperationRequest) returns (MakeBillPaymentOperationResponse);
  rpc MakeCashWithdrawalOperation (MakeCashWithdrawalOperationRequest) returns (MakeCashWithdrawalOperationResponse);
}